### ISSUE TITLE
`const T * Array<T>::fortran_vec (void) const` was removed in Octave 9

### ISSUE THREAD

**Issue by mmuetzel** [gnu-octave/pkg-control#7]
`Array<T>::fortran_vec` has been deprecated in Octave 7 and will be removed in Octave 9 (likely to be released in early 2025).

Please, consider applying a patch similar to the one in MXE Octave:
https://hg.octave.org/mxe-octave/file/tip/src/of-control-2-octave-9-compat.patch

Octave core tries to add deprecation attributes for functions that are slated for removal (like `fortran_vec` here). But the build rules of the control package disable these deprecation warnings. That means that users (and developers) of the control package will be surprised by the "sudden" removal of these functions (instead of being notified about it about two years in advance).
Please, consider to also remove the flag that silences these deprecation notices. E.g, with a change similar to the one in MXE Octave:
https://hg.octave.org/mxe-octave/file/tip/src/of-control-1-deprecated-warnings.patch

**Comment by ttl-octave** [comment]
Thank you very much for this heads up. I will release a fixed version ASAP.
Did you mean that `Array<T>::fortran_vec` will be removed in Octave 10?

**Comment by mmuetzel** [comment]
Features are usually deprecated in core Octave during two major versions. Afaict, `Array<T>::fortran_vec const` started to be deprecated in Octave 7:
https://hg.savannah.gnu.org/hgweb/octave/rev/7faff48840eb

Octave 8 was the second major version for which it is deprecated.
It is removed on the current default branch of core Octave which will eventually (likely early 2025) be released as Octave 9.

**Comment by mmuetzel** [comment]
Oops. You are right.

I'm now doubting what I thought I remember why we added that patch in MXE Octave:
I'm pretty sure we needed that patch when we started to build Octave from its default branch after that function overload was removed in Octave 9 (building in MXE Octave that is). That function was finally removed in this changeset:
https://hg.savannah.gnu.org/hgweb/octave/rev/db8735ee84da#l4.1

I'm also confused how it is apparently still possible to build the control package with Octave 9 without the patch from MXE Octave. 😕
Maybe, different compiler versions are using different implicit deduction rules, and you end up using `T * Array<T>::fortran_vec (void)` instead of `T * Array<T>::fortran_vec (void) const` (note the difference in the `const` qualifier) now? Or maybe, that depends on which platform is being targeted? (Or I'm misremembering why we needed that patch in MXE Octave...)

Fwiw, that patch was added in MXE Octave with this changeset:
https://hg.octave.org/mxe-octave/rev/8ea9085d1f80
That was one day after the function was removed on the default branch of Octave. So, the timeline matches what I think I'm remembering... 🤔

**Comment by ttl-octave** [comment]
Thanks for digging into this. `fortran_vec ()` is used many times in the wrapper functions for the fortran routines (341 matches).

But I just found [this commit](https://github.com/gnu-octave/pkg-control/commit/b5f513016978e97d9f4c71e67b62f0c4ca77f8fa) which brings some light into all this. I fixed a build error for Octave 9 with this commit (without knowing that `const` wasn't the issue in the first place). Obviously, `sl_ab08nd.cc` is the only function that defined the arrays as `const` and therefore the only one that needed the patch. 

I will re-add the `const` attribute and will use `data ()` instead of `fortran_vec ()`.
